syntax/printer_test.go

```diff
@@ -616,6 +616,24 @@ var printTests = []printCase{
 		"`declare`",
 		"$(declare)",
 	},
+	{
+		"(\n(foo >redir)\n)",
+		"(\n\t(foo >redir)\n)",
+	},
+	{
+		"( (foo) )",
+		"( (foo))",
+	},
+	{
+		"( {foo}; bar; )",
```

Comment:
Did you mean to use (foo) here? That's what we want to test AFAICT

It could also be nice to include a test for the other form of ambiguity when we add a newline:

( ((foo)); bar)

Reply:
actually, will include fix :)

Reply:
@mvdan please see the latest [commit](https://github.com/mvdan/sh/pull/869/commits/8e4f46fc45ff1319f3cad818f5a78cd53ab2c17a) for my attempt to fix the scenario above.